keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -0,0 +1,730 @@
+  anonymousAccessMode: [6]
+  bucketClassName: [7]
+  bucketRequest: [8]
+  allowedNamespaces: [9]
+    - name:
+  protocol: [10]
+    name:
+    version:
+    azureBlob:
+      containerName:
+      storageAccount:
+    s3:
+      endpoint:
+      bucketName:
+      region:
+      signatureVersion:
+    gs:
+      bucketName:
+      privateKeyName:
+      projectId:
+      serviceAccount:
+  parameters: [11]
+status:
+  bucketAvailable: [12]
+```
+
+1. `name`: When created by COSI, the `Bucket` name is generated in this format: _"<BR.namespace*>-<BR.name*>-<uuid>"_. If an admin creates a `Bucket`, as is necessary for brownfield access, they can use any name. The uuid is unique within a cluster. `*` the first 100 characters of the namespace and the name are used.
+    cosi.io/provisioner:
+    cosi.io/bucket-prefix:
+2. `labels`: added by COSI. The "cosi.io/provisioner" key's value is the provisioner name. The "cosi.io/bucket-prefix" key's value is the `BucketRequest.bucketPrefix` value when specified. Characters that do not adhere to [Kubernetes label conventions](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set) will be converted to ‘-’.
+3. `finalizers`: added by COSI to defer `Bucket` deletion until backend deletion ops succeed. Implementation may add one finalizer for the BR and one for the BA.
+4. `provisioner`: (optional) The provisioner field as defined in the `BucketClass`. If empty then there is no driver/sidecar, thus the admin had to create the `Bucket` and `BucketAccess` instances.
+5. `retentionPolicy`: Prescribes the outcome when the `BucketRequest` is deleted.
+   - _Retain_: (default) the `Bucket` and its data are preserved. The `Bucket` can potentially be reused.
+   - _Delete_: the bucket and its contents are destroyed.
+> Note: the `Bucket`'s retention policy is set to "Retain" as a default. Exercise caution when using the "Delete" retention policy as the bucket content will be deleted.
+> Note: a `Bucket` is not deleted if it is bound to a `BucketRequest` or `BucketAccess`.
+6. `anonymousAccessMode`: a string specifying *uncredentialed* access to the Bucket.  This is applicable to cases where the bucket objects are intended to be publicly readable and/or writable. One of:
+   - "private": Default, disallow uncredentialed access to the backend storage.
+   - "publicReadOnly": Read only, uncredentialed users can call ListBucket and GetObject.
+   - "publicWriteOnly": Write only, uncredentialed users can only call PutObject.
+   - "publicReadWrite": Read/Write, same as `ro` with the addition of PutObject being allowed.
+7. `bucketClassName`: (optional for greenfield, omitted for brownfield) name of the associated bucket class. 
+8. `bucketRequest`: an `objectReference` containing the name, namespace and UID of the associated `BucketRequest`.
+9. `allowedNamespaces`: a list of namespaces that are permitted to either create new buckets or to access existing buckets.
```

Comment:
Networking folks are also looking for ways to address this problem in GatewayClass. In the future, we may want to consider aligning with whatever they land on for consistency across k8s.